app/src/main/java/org/septa/android/app/transitview/TransitViewFragment.java

```diff
@@ -0,0 +1,290 @@
+                newFragment.setTargetFragment(TransitViewFragment.this, 3);
+                newFragment.show(ft, "line_picker");
+            }
+        });
+
+        // reset selection
+        resetView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                firstRoute = null;
+                secondRoute = null;
+                thirdRoute = null;
+
+                firstRoutePicker.setText(R.string.transitview_line_picker_first_text);
+                secondRoutePicker.setText(R.string.transitview_line_picker_second_text);
+                thirdRoutePicker.setText(R.string.transitview_line_picker_third_text);
+
+                firstRoutePicker.setCompoundDrawablesWithIntrinsicBounds(null, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+                secondRoutePicker.setCompoundDrawablesWithIntrinsicBounds(null, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+                thirdRoutePicker.setCompoundDrawablesWithIntrinsicBounds(null, null, ContextCompat.getDrawable(getContext(), R.drawable.ic_line_picker), null);
+
+                disableView(secondRoutePicker);
+                disableView(thirdRoutePicker);
+                disableView(queryButton);
+            }
+        });
+
+        queryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                goToTransitViewResults();
+            }
+        });
+
+        return rootView;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (resultCode == TransitViewLinePickerFragment.SUCCESS) {
+            switch (requestCode) {
+                case 1:
+                    selectFirstRoute((RouteDirectionModel) data.getSerializableExtra(TransitViewLinePickerFragment.ROUTE_DIRECTION_MODEL));
+                    break;
+                case 2:
+                    selectSecondRoute((RouteDirectionModel) data.getSerializableExtra(TransitViewLinePickerFragment.ROUTE_DIRECTION_MODEL));
+                    break;
+                case 3:
+                    selectThirdRoute((RouteDirectionModel) data.getSerializableExtra(TransitViewLinePickerFragment.ROUTE_DIRECTION_MODEL));
+                    break;
+                default:
+                    Log.e(TAG, "Invalid request code returned from TransitViewLinePicker");
+                    break;
+            }
+
+        } else {
+            Log.e(TAG, "Unsuccessful request code from TransitViewLinePicker #" + requestCode);
+        }
+    }
+
+    @Override
+    public void selectFirstRoute(RouteDirectionModel route) {
```

Comment:
Seems like these 3 methods could be 1 method with parameters.  Or maybe just the stuff around the bullets?

Reply:
I suppose it could be... I thought to separate it out since those methods (selectFirstRoute(), selectSecond(), etc.) are part of a listener both the TVResultsActivity and the TVFragment (picker) call, and both have specific logic around what views get enabled / disabled afterwards. Rather than doing multiple route ID comparisons each time a route was added / removed I thought this might be cleaner?

Reply:
Ok so maybe just make lines 207-210 a method, pass in a reference to your route picker, use that method in your 3 other methods.

Reply:
resolved with https://github.com/septadev/SEPTA-Android/pull/182/commits/90c0e46f4a4635dccba2ba95485f71ae46514fce